udp-client/src/lib.rs

```diff
@@ -63,10 +63,17 @@ pub struct UdpConfig {
 
 impl NewConnectionConfig for UdpConfig {
     fn new() -> Result<Self, ClientError> {
+        // Use UNSPECIFIED for production validators to bind to all interfaces
+        // Use LOCALHOST only in dev/test context to avoid port conflicts in CI
+        #[cfg(not(feature = "dev-context-only-utils"))]
+        let bind_ip = std::net::IpAddr::V4(std::net::Ipv4Addr::UNSPECIFIED);
+        #[cfg(feature = "dev-context-only-utils")]
```

Comment:
This is necessary but not sufficient to fix the flaky port collision issue with unit tests as it must be enabled by the test to use the feature in its dev-dependencies.  But I think we can get this in first as and address the test part in a separate PR as there are quite a few crates need that change.

Reply:
CI should work mostly fine even if it binds to all interfaces. I agree we need to merge this ASAP.